Adding omni_userLogin rpc method

silva-fj opened the pull request:
This PR introduces a new rpc method to login (get JWTs) to be able to make subsequent requests to the worker, as well as some small refactorings.

There following structs have been introduced: `UserId`, `UserAuth` and `ClientAuth` to handle authentication and simplify the API params.

TODO: 
- [x] Update `heima_post_login`'s path in pumpx api module

silva-fj committed:
refactor: simplify serde format for enum serialization

Remove explicit serde tag/content attributes from IdentitySerde and
HeimaMultiSignature enums to use Rust's default enum serialization
format. This changes the JSON structure from:
{"type": "Variant", "data": "value"}
to:
{"Variant": "value"}

Updated corresponding tests to reflect the new serialization format.
This simplifies the API and makes the JSON more concise while
maintaining the same functionality.

silva-fj committed:
refactor: add TryFrom impl for OmniAuth conversion

Add TryFrom<OmniAuthSerde> for OmniAuth implementation to handle
conversion from serde format to main enum

silva-fj committed:
feat: add UserInfo struct and user_login RPC method (WIP)

silva-fj committed:
feat: add omni_userlogin RPC method module

silva-fj committed:
feat: implement UserLogin native task handling (WIP)

silva-fj committed:
initial auth refactoring (WIP)

silva-fj committed:
refactor: remove UserLogin native task variant

silva-fj committed:
refactor: rename auth types for clarity

This improves code readability by using more descriptive type names
that better reflect their purpose in the authentication system.

silva-fj committed:
feat: implement omni_userLogin RPC method (WIP)

silva-fj committed:
feat(rpc): add chrono dependency for user login method

silva-fj committed:
feat(rpc): implement JWT token generation in user login

Implement JWT token generation for user authentication in the
omni_userLogin RPC method with proper error handling and validation.

silva-fj committed:
fix(auth): change TryFrom error type to unit type

silva-fj committed:
refactor(auth): remove unused UserInfo struct

silva-fj committed:
update TODOs

silva-fj committed:
Merge branch 'dev' into p-1543-create-new-rpc-method-omni_userlogin-to-replace

silva-fj committed:
Merge branch 'dev' into p-1543-create-new-rpc-method-omni_userlogin-to-replace

silva-fj committed:
removing pumpx specific user_id

silva-fj committed:
updating test name

silva-fj committed:
fix build

Kailai-Wang commented on tee-worker/omni-executor/executor-primitives/src/auth.rs:
```diff
@@ -92,12 +91,45 @@ pub enum OmniAuth {
+
+#[derive(Deserialize, Encode, Decode, Clone, PartialEq, Eq, Debug)]
+pub enum UserAuth {
+	Web3(HeimaMultiSignature),
```
I think it's a good opportunity to rewrite (flattern) it a bit, no `Web3` wrapper, just `Evm`, `Substrate`, `Solana`... variants
- to be consistent with `UserId`
- to avoid confusions like `ecdsa` vs `ethereum`, actually `ecdsa` is in substrate context, ethereum signatures might be wrapped with EIP191

silva-fj replied:
Yeah it sounds good 👍🏼

Kailai-Wang commented on tee-worker/omni-executor/executor-primitives/src/auth.rs:
```diff
@@ -1,23 +1,26 @@
-#[serde(tag = "type", content = "data")]
-pub enum IdentitySerde {
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq, Encode, Decode)]
+pub enum UserId {
```
Shall we use snake_case everywhere? (to be consistent with other rpc fields)? The first char will become small letter by default I assume

silva-fj committed:
feat: implement pumpx client authentication in omni_userLogin

- Add ClientAuth enum with Pumpx variant for google_code and invite_code
- Extend UserLoginParams to include optional client_auth field
- Implement pumpx-specific authentication flow:
  * Verify user exists via get_account_user_id API call
  * Validate google_code through user_connect API call
  * Store access token in PumpxJwtStorage for future use
- Generate both id_token and access_token for authenticated users
- Make error codes public for pumpx API failures
- Add proper error handling for missing client auth data

silva-fj committed:
feat: add snake_case serialization to UserId enum

silva-fj committed:
Merge remote-tracking branch 'origin/dev' into p-1543-create-new-rpc-method-omni_userlogin-to-replace

silva-fj committed:
export individual signatures from heima_multi_signature

silva-fj committed:
feat(auth): split Web3 UserAuth into specific variants

Refactor the UserAuth enum to replace the generic Web3 variant with
specific variants for Substrate, EVM, Solana, and Bitcoin signatures.

silva-fj committed:
fix(logging): restore debug log for user_connect call

silva-fj committed:
refactor: rename PumpxJwtStorage to HeimaJwtStorage

silva-fj committed:
adding post login placeholders and refactoring names

silva-fj committed:
Merge branch 'dev' into p-1543-create-new-rpc-method-omni_userlogin-to-replace

silva-fj committed:
cleaning up comments

silva-fj committed:
Merge branch 'dev' into p-1543-create-new-rpc-method-omni_userlogin-to-replace

silva-fj committed:
feat: standardize serde serialization for auth enums

- Add tagged union serialization to UserId, UserAuth, ClientAuth, and
  SubstrateSignature enums using #[serde(tag = "type", content = "value")]
- Update JSON format from {"email": "value"} to
  {"type": "email", "value": "value"}
- Add snake_case rename_all attribute to ClientAuth enum for consistency
- Update test case to match new JSON serialization format

This change provides consistent JSON representation across all auth-related
enums, making the API more predictable and easier to consume by clients.

silva-fj committed:
Merge remote-tracking branch 'origin/dev' into p-1543-create-new-rpc-method-omni_userlogin-to-replace

silva-fj committed:
fixing imports

silva-fj committed:
feat: add wildmeta client_id

silva-fj committed:
feat: add SHA256-based omni account derivation with client_id

Replace Blake2 hash with SHA256 for omni account generation using
pattern: sha256(client_id + user_id_type + user_id_value).

silva-fj committed:
Merge remote-tracking branch 'origin/dev' into p-1543-create-new-rpc-method-omni_userlogin-to-replace

silva-fj committed:
updating lock files

silva-fj committed:
refactor: update omni account calls to use client-specific IDs

Replace generic to_omni_account() calls with client-specific
to_omni_account_with_client_id() throughout the codebase.

silva-fj committed:
refactor: rename omni_getMessageCode to omni_getWeb3SignInMessage

Update RPC method name to better reflect its purpose of generating
Web3 sign-in messages for authentication.

silva-fj committed:
refactor: update OmniAuth::Web3 to include client_id parameter

silva-fj committed:
fixing fmt issue

silva-fj committed:
feat: add heima_post_login API method to PumpX client

silva-fj committed:
feat: integrate heima_post_login API into user login flow

silva-fj committed:
Merge branch 'dev' into p-1543-create-new-rpc-method-omni_userlogin-to-replace

silva-fj committed:
update get sign in message response

silva-fj committed:
Merge branch 'dev' into p-1543-create-new-rpc-method-omni_userlogin-to-replace

Kailai-Wang committed:
update path

Kailai-Wang committed:
Fixes and updates

Kailai-Wang committed:
try to disable oa CI test
